### PR TITLE
Fix dependency checking

### LINKED ISSUE #835: CEkit builds log missing dependencies errors as DEBUG instead of ERROR
**Describe the bug**
CEkit builds log missing dependencies errors as `DEBUG`  instead of `ERROR`
```
2023-01-17 11:58:24,234 resource.py:589        DEBUG [Errno 2] No such file or directory: '/usr/bin/brew'
```

**To reproduce**
Install CEkit and run a build without installing required dependencies [1]

**Expected behavior**
Cekit should log errors related to missing dependencies or commands as `ERROR` instead of `DEBUG`.

**Logs**
```
[k@fedora operator]$ cekit -v build docker 
2023-01-17 11:58:24,115 cli.py:440        DEBUG Running version 4.5.0
2023-01-17 11:58:24,116 cli.py:453        DEBUG Configuring CEKit...
2023-01-17 11:58:24,116 cli.py:470        DEBUG Removing dirty directory: 'target/image/modules'
2023-01-17 11:58:24,117 cli.py:470        DEBUG Removing dirty directory: 'target/repo'
2023-01-17 11:58:24,117 tools.py:457        INFO  You are running on known platform: Fedora Linux 37 (Workstation Edition)
2023-01-17 11:58:24,117 builder.py:27         DEBUG Checking CEKit core dependencies...
2023-01-17 11:58:24,117 tools.py:513        DEBUG Checking if 'git' dependency is provided...
2023-01-17 11:58:24,117 tools.py:575        DEBUG CEKit dependency 'git' provided via the '/usr/bin/git' executable.
2023-01-17 11:58:24,117 tools.py:548        DEBUG All dependencies provided!
2023-01-17 11:58:24,118 tools.py:611        WARNING The certifi library (https://certifi.io/) was found, depending on the operating system configuration this may result in certificate validation issues
2023-01-17 11:58:24,118 tools.py:617        WARNING Certificate Authority (CA) bundle in use: '/etc/pki/tls/certs/ca-bundle.crt'
2023-01-17 11:58:24,180 builder.py:88         INFO  Generating files for docker engine
2023-01-17 11:58:24,180 base.py:102        INFO  Initializing image descriptor...
2023-01-17 11:58:24,181 builder.py:108        DEBUG Checking CEKit generate dependencies...
2023-01-17 11:58:24,181 tools.py:513        DEBUG Checking if 'odcs-client' dependency is provided...
2023-01-17 11:58:24,181 tools.py:519        DEBUG Required CEKit library 'odcs-client' was found as a 'odcs' module!
2023-01-17 11:58:24,181 tools.py:513        DEBUG Checking if 'brew' dependency is provided...
2023-01-17 11:58:24,181 tools.py:513        DEBUG Checking if 'skopeo' dependency is provided...
2023-01-17 11:58:24,181 tools.py:548        DEBUG All dependencies provided!
2023-01-17 11:58:24,182 base.py:127        DEBUG Removing old target directory
2023-01-17 11:58:24,182 tools.py:120        DEBUG Reading descriptor from 'image.yaml' file...
2023-01-17 11:58:24,192 base.py:265        DEBUG Downloading module repository: 'modules'
2023-01-17 11:58:24,193 resource.py:226        INFO  Copying resource 'modules'...
2023-01-17 11:58:24,193 resource.py:395        DEBUG Copying repository from '/home/k/work/prod/amqstreams-1-openshift-image/operator/modules' to 'target/repo/modules'.
2023-01-17 11:58:24,193 tools.py:120        DEBUG Reading descriptor from '/home/k/work/prod/amqstreams-1-openshift-image/operator/target/repo/modules/operator/module.yaml' file...
2023-01-17 11:58:24,198 base.py:284        DEBUG Adding module 'operator', path: 'target/repo/modules/operator'
2023-01-17 11:58:24,199 image.py:426        DEBUG Preparing module 'operator' required by 'amq7/amq-streams-rhel8-operator'.
2023-01-17 11:58:24,201 base.py:315        DEBUG Copying module 'operator' required by 'amq7/amq-streams-rhel8-operator'.
2023-01-17 11:58:24,201 base.py:324        DEBUG Copying module 'operator' to: 'target/image/modules/operator'
2023-01-17 11:58:24,205 docker.py:25         INFO  Handling artifacts for docker...
2023-01-17 11:58:24,205 resource.py:226        INFO  Copying resource 'cluster-operator-dist.zip'...
2023-01-17 11:58:24,206 resource.py:301        DEBUG Using md5 checksum to fetch artifacts from cacher
2023-01-17 11:58:24,206 resource.py:311        DEBUG Using cache url 'http://cache.host.com/fetch?md5=8eb508ff01f6073de982324eb8271e3b'
2023-01-17 11:58:24,206 tools.py:39         DEBUG Downloading from 'http://cache.host.com/fetch?md5=8eb508ff01f6073de982324eb8271e3b' as /tmp/cache/19d88ff8-bf8b-4617-b7db-17ff153e1fef
2023-01-17 11:58:24,233 resource.py:570        DEBUG <urlopen error [Errno -2] Name or service not known>
2023-01-17 11:58:24,233 resource.py:571        WARNING Could not download 'cluster-operator-dist.zip' artifact using cacher
2023-01-17 11:58:24,233 resource.py:578        DEBUG Trying to download artifact 'cluster-operator-dist.zip' from Brew directly
2023-01-17 11:58:24,233 tools.py:197        DEBUG Getting brew details for an artifact with '8eb508ff01f6073de982324eb8271e3b' md5 sum
2023-01-17 11:58:24,233 tools.py:207        DEBUG Executing '/usr/bin/brew call --json-output listArchives checksum=8eb508ff01f6073de982324eb8271e3b type=maven'.
2023-01-17 11:58:24,234 resource.py:589        DEBUG [Errno 2] No such file or directory: '/usr/bin/brew'
2023-01-17 11:58:24,234 resource.py:590        WARNING Could not download artifact 'cluster-operator-dist.zip' from Brew
2023-01-17 11:58:24,234 resource.py:251        WARNING Cekit is not able to fetch resource 'cluster-operator-dist.zip' automatically. Please use cekit-cache command to add this artifact manually.
2023-01-17 11:58:24,234 cli.py:500        ERROR ("Error copying resource: 'cluster-operator-dist.zip'. See logs for more info.", CekitError('Artifact cluster-operator-dist.zip could not be found'))
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/cekit/descriptor/resource.py", line 249, in guarded_copy
    self._copy_impl(target)
  File "/usr/lib/python3.11/site-packages/cekit/descriptor/resource.py", line 594, in _copy_impl
    raise CekitError("Artifact {} could not be found".format(self.name))
cekit.errors.CekitError: Artifact cluster-operator-dist.zip could not be found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/cekit/cli.py", line 486, in run
    command.execute()
  File "/usr/lib/python3.11/site-packages/cekit/builder.py", line 69, in execute
    self.generate()
  File "/usr/lib/python3.11/site-packages/cekit/builder.py", line 114, in generate
    self.generator.generate()
  File "/usr/lib/python3.11/site-packages/cekit/generator/base.py", line 184, in generate
    self.prepare_artifacts()
  File "/usr/lib/python3.11/site-packages/cekit/generator/docker.py", line 30, in prepare_artifacts
    artifact.copy(target_dir)
  File "/usr/lib/python3.11/site-packages/cekit/descriptor/resource.py", line 240, in copy
    self.cache.add(self)
  File "/usr/lib/python3.11/site-packages/cekit/cache/artifact.py", line 63, in add
    artifact.guarded_copy(artifact_file)
  File "/usr/lib/python3.11/site-packages/cekit/descriptor/resource.py", line 262, in guarded_copy
    raise CekitError(
cekit.errors.CekitError: ("Error copying resource: 'cluster-operator-dist.zip'. See logs for more info.", CekitError('Artifact cluster-operator-dist.zip could not be found'))
```

**Environment information:**
 - Operating system: fedora 37
 - CEKit version: 4.5.0

**Additional context**
[1] https://docs.cekit.io/en/latest/handbook/installation/dependencies.html?highlight=brew#docker-builder-dependencies

### PR DESCRIPTION
**Pull request by rnc**
Fixes #835

### CODE REVIEW

**Comment by rnc** [comment]
To fully resolve this I believe we need to remove explicit paths from within CEKit and also add a $HOME/bin to MacOS as a dummy brew executable.

File: cekit/tools.py [hunk 1]
```diff
@@ -608,6 +614,10 @@ def handle_core_dependencies(self) -> None:
         try:
             import certifi
 
+            # If the certificate bundle ends up resolving e.g. /usr/local/lib/python3.7/site-packages/certifi/cacert.pem
+            # rather than /etc/pki/tls/certs/ca-bundle.crt or /etc/ssl/certs/ca-bundle.crt this can cause issues
+            # hence printing the warning. A user workaround is to export REQUESTS_CA_BUNDLE to point to the correct
+            # certificates
```
Comment (goldmann) [inline cekit/tools.py:620]: I wonder if we could add this information to the warning message below as well.

Comment (goldmann) [reply cekit/tools.py:620]: This, or similar, that is.

Comment (rnc) [reply cekit/tools.py:620]: Added another log statement.

**Review by goldmann — APPROVED** [review]

**Pull request merged**